qiskit_experiments/library/characterization/cr_hamiltonian.py

```diff
@@ -177,12 +192,39 @@ def _default_experiment_options(cls) -> Options:
         options.sigma = 64
         options.risefall = 2
 
+        # These are automatically filled when backend is set
+        options.cr_channel = 0
+        options.dt = 1e-9
+        options.granularity = 1
+
         return options
 
-    def _build_cr_circuit(
-        self,
-        pulse_gate: circuit.Gate,
-    ) -> QuantumCircuit:
+    def _set_backend(self, backend: Backend):
+        super()._set_backend(backend)
+
+        # Extract control channel index
+        try:
```

Comment:
I implemented slightly different logic to check if value and new value are identical (then we don't need to raise warning). This is the case when we set backend multiple times.

Reply:
The warning is intended for when the backend doesn't have the required information to run the experiment. In these case I think you should get a warning or exception rather than using default values.

Eg. I'm not sure `dt` should be an experiment option when it is always expected to come from a backend property? Setting a custom value that is different to a backend will generally not work. In that sense I don't think it makes sense to set a default `dt=1e-9`. If a backend doesn't provide a dt you should raise an exception.

The other two options for granularity and cr_channel i'm not as familiar with so don't know if they make sense as experiment options or not. But looking at the code it seems CR channel isn't something that should be set by a user, since it is computed from the backend from the specified physical qubits of the experiment, so if a backend doesn't provide that information it should raise an error too. For granularity i'm not sure, since you extract from backend timing constraints it also looks like a backend property, but maybe there is a case for a user setting a custom value for this one?